### ISSUE TITLE
Wrong typename returned from server

### ISSUE THREAD

**Issue by nakeeon** [graphql-python/graphene#536]
I have an issue with data returned from the server (graphene and Django). I use apollo client and React.js on the front end. I have a component in which a graphql query is executes:
```
query getPositions {
    providerPositions {
      id
      title
    }
    businessTypes {
      id
      name
    }
  }
```

On the server:
```
class BusinessTypeType(DjangoObjectType):
    class Meta:
        model = BusinessTypeModel


class ServiceProviderType(DjangoObjectType):
    class Meta:
        model = ServiceProvider

    def resolve_business_types(self, args, context, info):
        return BusinessTypeModel.objects.all()

    def resolve_provider_positions(self, args, context, info):
        return ProviderPosition.objects.all()
```

It returns `BusinessTypeType` typname for the `providerPositions` which is wrong. `ProviderPositionType` is expected.
![947304dccf](https://user-images.githubusercontent.com/17749002/29831924-50da04d4-8cef-11e7-9ae8-ff4ad0d843db.jpg)

It cause a mess and mixing in the returned data.

**Comment by syrusakbary** [comment]
Hi @nakeeon,

Thanks for reporting the issue! Could be possible to have a full test case that reproduces your issue or a repo that I can clone?

**Comment by nakeeon** [comment]
@syrusakbary 
I solved the issue. Graphene is fine. The problem was in the Apollo client, because  Apollo cannot determine the IDs to use for object. There is a `dataIdFromObject` function for that purpose. I took this peace of code from the Apollo tutorial:
```
const client = new ApolloClient({
  dataIdFromObject: o => o.id
});
```

But I have several Django models with the same ids, so it caused the problem with typename of objects.

Changing the code to this is solved the problem:

```
const client = new ApolloClient({
  dataIdFromObject: o => `${o.__typename}-${o.id},`
});
```

Sorry for disturbing.
